src/main/java/picard/illumina/SortedBasecallsConverter.java

```diff
@@ -234,38 +224,33 @@ private synchronized void addRecord(final String barcode, final CLUSTER_OUTPUT_R
     protected void awaitTileProcessingCompletion() throws IOException {
         tileReadExecutor.shutdown();
         // Wait for all the read threads to complete before checking for errors
-        ThreadPoolExecutorUtil.awaitThreadPoolTermination("Reading executor", tileReadExecutor, Duration.ofMinutes(5));
-
-        // Check for reading errors
-        if (tileReadExecutor.hasError()) {
-            interruptAndShutdownExecutors(tileReadExecutor, tileWriteExecutor);
-        }
+        awaitExecutor(tileReadExecutor);
 
         int tileProcessingIndex = 0;
-
+        ThreadPoolExecutorWithExceptions tileWriteExecutor = null;
         while (tileProcessingIndex < tiles.size()) {
-            if (tileWriteJobs.get() == 0) {
-                completedWork.get(tiles.get(tileProcessingIndex)).forEach(tileWriteExecutor::submit);
+                awaitExecutor(tileWriteExecutor);
+                ThreadPoolExecutorWithExceptions finalTileWriters = new ThreadPoolExecutorWithExceptions(numThreads);
```

Comment:
Can this be final?

Reply:
Actually the intermediate variable isn't even needed since we aren't using it in a lambda